smartsim/entity/dbobject.py

```diff
@@ -118,17 +120,20 @@ def _enumerate_devices(self) -> t.List[str]:
 
         if self.device == "GPU" and self.devices_per_node > 1:
             return [
-                f"{self.device}:{str(device_num)}"
+                f"{self.device}:{str(device_num + self.first_device)}"
                 for device_num in range(self.devices_per_node)
             ]
```

Comment:
Pedantic style thing: You can save the redundant binary ops by supplying a second arg to range. Also, not from this PR, but there is no need for the explicit cast to `str`
```py
[
    f"{self.device}:{device_num}"
    for device_num in range(self.first_device, self.first_device + self.devices_per_node)
]
```